### DISCUSSION TITLE
GraphQL permission

### DISCUSSION THREAD

**Discussion by gaetandezeiraud** [OrchardCMS/OrchardCore#10679] [Q&A] [answered]
Hi! 
I want to use Orchard like an Headless CMS. With a NextJS frontend. It is working good, here my repo https://github.com/Brouilles/orchard-blog-nextjs-frontend. But, for it to work. I need to put Anonymous with permission to use GraphQL and the BlogPost. Of course, not ideal for a real world situation.

My NextJS call, on the server side of Next, Orchard from `getServerSideProps` ([here](https://github.com/Brouilles/orchard-blog-nextjs-frontend/blob/main/orchard-blog-nextjs-frontend/pages/index.js)). My question is, how I can get a token from Orchard to use in my NextJS backend? I can add a `authorization Bearer` in the request (code commented in my repo) but I don't know how I can get from Orchard Admin Dashboard.

Hope it is clear and you can help me!
Thanks.

**Comment by gaetandezeiraud** [comment]
I found this very good article https://medium.com/swlh/orchard-core-open-id-connect-and-graphql-f003a222260a. But I have always a question. The access token expires shortly. So I need to request it each time I need to call graphQL endpoint?

**Comment by Skrypt** [comment, marked as answer]
There are no mutations yet in the GraphQL endpoint. To manage your content items you need to use the REST API endpoint. You could use OpenApi/Swagger for that matter.
